server/graphql/v2/mutation/VendorMutations.ts

```diff
@@ -160,20 +160,27 @@ const vendorMutations = {
         await setTaxForm(vendor, args.vendor.vendorInfo.taxFormUrl, new Date().getFullYear());
       }
 
-      if (args.vendor.payoutMethod && !args.vendor.payoutMethod?.id) {
-        const existingPayoutMethods = await vendor.getPayoutMethods();
-        if (!isEmpty(existingPayoutMethods)) {
-          existingPayoutMethods.map(pm => pm.update({ isSaved: false }));
+      if (args.vendor.payoutMethod) {
+        const existingPayoutMethods = await vendor.getPayoutMethods({ where: { isSaved: true } });
+        if (!args.vendor.payoutMethod.id) {
+          if (!isEmpty(existingPayoutMethods)) {
+            existingPayoutMethods.map(pm => pm.update({ isSaved: false }));
+          }
+
+          await models.PayoutMethod.create({
+            ...pick(args.vendor.payoutMethod, ['name', 'data', 'type']),
+            CollectiveId: vendor.id,
+            CreatedByUserId: req.remoteUser.id,
+            isSaved: true,
+          });
+        } else {
+          await Promise.all(
+            existingPayoutMethods
+              .filter(pm => pm.id !== args.vendor.payoutMethod.id)
+              .map(pm => pm.update({ isSaved: false })),
```

Comment:
Missing an ID decode

```suggestion
            const payoutMethodId = idDecode(args.vendor.payoutMethod.id, 'payout-method');
            existingPayoutMethods
              .filter(pm => pm.id !== payoutMethodId)
              .map(pm => pm.update({ isSaved: false })),
```

Also, shouldn't we update the other info for payout method here? Something like:

```es6
const payoutMethodToUpdate = existingPayoutMethods.find(pm => pm.id === payoutMethodId);
await payoutMethodToUpdate.update(pick(args.vendor.payoutMethod, ...));
```

Reply:
Not really. We shouldn't update payout methods. This conditional is only making sure there's a single Payout Method per vendor available.